packages/protocol/tipmanager/tipmanager.go

```diff
@@ -246,178 +255,29 @@ func (t *TipManager) isPastConeTimestampCorrect(block *booker.Block) (timestampV
-			blockWalker.Push(block)
-			return false
-		}
-
-		markerWalker.Push(markerPreviousBlockPair{Marker: markers.NewMarker(sequenceID, index), PreviousBlock: block})
-		return true
-	})
+	return
 }
 
-func (t *TipManager) checkPair(pair markerPreviousBlockPair, blockWalker *walker.Walker[*booker.Block], markerWalker *walker.Walker[markerPreviousBlockPair], minSupportedTimestamp time.Time) (timestampValid bool) {
-	block, exists := t.engine.Tangle.BlockFromMarker(pair.Marker)
-	if !exists {
-		return false
-	}
-
-	// marker before minSupportedTimestamp
-	if block.IssuingTime().Before(minSupportedTimestamp) {
-		// marker before minSupportedTimestamp
-		if !t.blockAcceptanceGadget.IsMarkerAccepted(pair.Marker) {
-			// if not accepted, then incorrect
-			return false
-		}
-		// if closest past marker is accepted and before minSupportedTimestamp, then need to walk block past cone of the previously marker block
-		blockWalker.Push(pair.PreviousBlock)
-		return true
-	}
-	// accepted after minSupportedTimestamp
-	if t.blockAcceptanceGadget.IsMarkerAccepted(pair.Marker) {
-		return true
-	}
-
-	// unaccepted after minSupportedTimestamp
-	sequence, exists := t.engine.Tangle.Booker.Sequence(pair.Marker.SequenceID())
-	if !exists {
-		return false
-	}
-
-	// check oldest unaccepted marker time without walking sequence DAG
-	firstUnacceptedMarker := findFirstUnacceptedMarker(sequence, t.engine.Tangle.BlockCeiling, t.engine.FirstUnacceptedMarker)
-	if timestampValid = t.processMarker(pair.Marker, minSupportedTimestamp, firstUnacceptedMarker); !timestampValid {
-		return false
-	}
-
-	// If there is an accepted marker before the oldest unaccepted marker, and it's older than minSupportedTimestamp, need to walk block past cone of firstUnacceptedMarker.
-	if sequence.LowestIndex() < firstUnacceptedMarker.Index() {
-		acceptedMarker, acceptedMarkerExists := t.engine.Tangle.BlockFloor(markers.NewMarker(sequence.ID(), firstUnacceptedMarker.Index()-1))
-		if !acceptedMarkerExists {
-			acceptedMarker = markers.NewMarker(sequence.ID(), sequence.LowestIndex())
-		}
-
-		if t.isMarkerOldAndAccepted(acceptedMarker, minSupportedTimestamp) {
-			firstUnacceptedMarkerBlock, blockExists := t.engine.Tangle.Booker.BlockFromMarker(firstUnacceptedMarker)
-			if !blockExists {
-				return false
-			}
-			blockWalker.Push(firstUnacceptedMarkerBlock)
-		}
-	}
-
-	// process markers from different sequences that are referenced by current marker's sequence, i.e., walk the sequence DAG
-	sequence.ReferencedMarkers(pair.Marker.Index()).ForEach(func(sequenceID markers.SequenceID, index markers.Index) bool {
-		// Ignore Marker(_, 0) as it sometimes occurs in the past marker cone. Marker mysteries.
-		// This should not be necessary anymore?
-		if index == 0 {
-			fmt.Println("(time: ", time.Now(), ") ignore Marker(_, 0) during tip fishing")
+func (t *TipManager) checkBlockRecursive(block *booker.Block, minSupportedTimestamp time.Time) (timestampValid bool) {
```

Comment:
Is this recursion guaranteed to not overflow? Can we refactor this into a for-loop?

Reply:
It is not guaranteed, no. Implementing it iteratively is definitely possible, we will need to introduce a "breadcrumb" accumulator to remember and mark the path we are travelling from, while still exploring the graph in other directions whenever a certain path ends in a `true` return.
In go, the maximum allowed stack-depth is defined in bytes, rather than in # of frames. The limit is currently set at 1GB for 64 bit and 250MB for 32 bit architectures respectively.
We can expect very deep stacks whenever acceptance stop for a significant amount of time, but in those cases, the depth of this call stack is the least of our concerns: as we are not evicting any object from memory whenever acceptance stops.
For this reason I don't think this is an actual threat to our runtime, as we are way more likely to incur in OOM exceptions because of the objects cache before hitting the maximum call stack size.